src/main/java/com/ldtteam/structures/blueprints/v1/Blueprint.java

```diff
@@ -229,23 +303,154 @@ public Blueprint setMissingMods(String... missingMods)
+    {
+        final List<Tuple<BlockPos, NBTTagCompound>> list = new ArrayList<>();
+        for (short x = 0; x < this.sizeX; x++)
+        {
+            for (short y = 0; y < this.sizeY; y++)
+            {
+                for (short z = 0; z < this.sizeZ; z++)
+                {
+                    list.add(new Tuple<>(new BlockPos(x,y,z), entities[y][z][x]));
+                }
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Get a list of all blockInfo objects in the blueprint.
+     * @return a list of all blockinfo (position, blockState, tileEntityData).
+     */
+    public final List<BlockInfo> getBlockInfoAsList()
+    {
+        final List<BlockInfo> list = new ArrayList<>();
+        for (short x = 0; x < this.sizeX; x++)
+        {
+            for (short y = 0; y < this.sizeY; y++)
+            {
+                for (short z = 0; z < this.sizeZ; z++)
+                {
+                    final BlockPos tempPos = new BlockPos(x, y, z);
+                    final short value = structure[y][z][x];
+                    final IBlockState state = palette.get(value & 0xFFFF);
+                    list.add(new BlockInfo(tempPos, state, tileEntities[y][z][x]));
+                }
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Rotate the structure depending on the direction it's facing.
+     *
+     * @param rotation  times to rotateWithMirror.
+     * @param pos the pos to rotateWithMirror it around.
+     * @param mirror    the mirror.
+     */
+    public BlockPos rotateWithMirror(final Rotation rotation, final BlockPos pos, final Mirror mirror)
+    {
+        final BlockPos resultSize = transformedBlockPos(sizeX, sizeY, sizeZ, mirror, rotation);
+        final short newSizeX = (short) resultSize.getX();
+        final short newSizeY = (short) resultSize.getY();
+        final short newSizeZ = (short) resultSize.getZ();
+
+        final short[][][] newStructure = new short[newSizeY][newSizeZ][newSizeX];
+        final NBTTagCompound[][][] newEntities = new NBTTagCompound[newSizeY][newSizeZ][newSizeX];
+        final NBTTagCompound[][][] newTileEntities = new NBTTagCompound[newSizeY][newSizeZ][newSizeX];
+
+        final List<IBlockState> palette = new ArrayList<>();
+        for (int i = 0; i < this.palette.size(); i++)
+        {
+            palette.add(i, this.palette.get(i).withRotation(rotation).withMirror(mirror));
+        }
+
+        this.palette = palette;
+
+        boolean foundAnchor = false;
+        BlockPos offset = pos;
+
+        for (short x = 0; x < this.sizeX; x++)
+        {
+            for (short y = 0; y < this.sizeY; y++)
+            {
+                for (short z = 0; z < this.sizeZ; z++)
+                {
+                    final BlockPos tempPos = transformedBlockPos(x, y, z, mirror, rotation);
+                    final short value = structure[y][z][x];
+                    final IBlockState state = palette.get(value & 0xFFFF);
```

Comment:
Don't know the purpose of mask (assuming it's needed), maybe a function or constant would be great

Reply:
It's needed because we're transforming the short to an int.

Reply:
Ok then, maybe some comment or integer.toShort.. I will add it in cleaning

Reply:
I think he uses unsigned short due to storage size reasons so we need it. Adding a comment would be nice though